packages/block-library/package.json

```diff
@@ -47,6 +47,7 @@
 		"@wordpress/date": "file:../date",
 		"@wordpress/deprecated": "file:../deprecated",
 		"@wordpress/dom": "file:../dom",
+		"@wordpress/editor": "file:../editor",
```

Comment:
The same issue is being tracked at https://github.com/WordPress/gutenberg/issues/32842. I believe that the best way to move forward is to move all the handling of the post from the `editor` store to entities in `@wordpress/core-data`. I believe it should be possible as of today, but I haven't tried.

Reply:
> The same issue is being tracked at #32842

Thanks for finding that one.

> move all the handling of the post from the `editor` store to entities in `@wordpress/core-data`. I believe it should be possible as of today

Yes, this is my suspicion too, but I don't know the entity-based APIs well enough to advise.

Reply:
```js
const MY_TYPE : 'post' | 'page' | ... = ...
const MY_ID : number = ...

// reading
const { comment_status } = useSelect( 'core' ).getEntityRecord( 'postType', MY_TYPE, MY_ID );
// writing
useDispatch( 'core' ).editEntityRecord( 'postType', MY_TYPE, MY_ID, { comment_status: 'open' } );
// saving
await useDispatch( 'core' ).saveEditedEntityRecord( 'postType', MY_TYPE, MY_ID );
```

This is the standard way approach for manipulating entities, as far as I'm aware. The only awkward bit that I see at the moment is that saving an edited entity seems to be all or nothing, so you may want to **not** call `saveEditedEntityRecord` and let the user manually save the whole post at the end when they are done — which is maybe a better UX anyway, no?

(Also, thanks to @adamziel for his [Core Data guides](https://developer.wordpress.org/block-editor/how-to-guides/data-basics/3-building-an-edit-form/) and for making it easier with types to find the expected shape of [Post](https://github.com/WordPress/gutenberg/blob/65550e38a68befd30250a05a05d76a7d42ebfc22/packages/core-data/src/entity-types/post.ts#L111-L115) and [CommentStatus](https://github.com/WordPress/gutenberg/blob/65550e38a68befd30250a05a05d76a7d42ebfc22/packages/core-data/src/entity-types/helpers.ts#L22).)

Reply:
Thanks @gziolo, #32842 is an interesting read.

> move all the handling of the post from the `editor` store to entities in `@wordpress/core-data`. I believe it should be possible as of today

Not totally sure I'm following. So for this PR, I'm using `editPost` from the editor store. This is defined here: https://github.com/WordPress/gutenberg/blob/65550e38a68befd30250a05a05d76a7d42ebfc22/packages/editor/src/store/actions.js#L130-L144

It should be easy enough to import `editEntityRecord` directly from `core-data`.

OTOH, `getCurrentPost` is defined in the `editor` store and directly relies on that store's state (`state.postId` and `state.postType`) -- which indeed seems to be fairly editor-specific. I'm not sure what we should do about those -- move them into `core-data`'s state?

Reply:
> `getCurrentPost` is defined in the `editor` store and directly relies on that store's state (`state.postId` and `state.postType`) -- which indeed seems to be fairly editor-specific. I'm not sure what we should do about those -- move them into `core-data`'s state?

I examined the `editor` store more closely. It looks like all handling for post/page/CPT is now fully controlled by `@wordpress/core-data`. The only part where `editor` store is helpful, as pointed out by @ockham, is when it automatically infers the post id and type. This is something that can be set through the [Block Context](https://github.com/WordPress/gutenberg/blob/trunk/docs/reference-guides/block-api/block-context.md). It looks like the Post Comments Form block has access to both post id and type:

https://github.com/WordPress/gutenberg/blob/d1a9f9717981dfd08429b9bd6d8f02252d2ae94a/packages/block-library/src/post-comments-form/block.json#L14

so this should work perfectly fine.

Reply:
The block context, as reminded by gziolo, is definitely your friend.

I wanted to get a personal feel for these APIs, so I got this working in my environment, preferring `useEntityProp` over the other functions that I previously mentioned:

```js
import { useEntityProp } from '@wordpress/core-data';

function MyBlockEdit( { context } ) {
  const { postId, postType } = context;
  const [ commentStatus, setCommentStatus ] = useEntityProp(
    'postType',
    postType,
    'comment_status'
  );
}
```

Reply:
Also surfacing `useEntityRecord`:

```js
import { useEntityRecord } from '@wordpress/core-data';

function PageTitleDisplay( { id } ) {
  const { record, isResolving } = useEntityRecord( 'postType', 'page', id );

  if ( isResolving ) {
    return 'Loading...';
  }

  return record.title;
}

// Rendered in the application:
// <PageTitleDisplay id={ 1 } />
```

Reply:
> The only awkward bit that I see at the moment is that saving an edited entity seems to be all or nothing, so you may want to not call saveEditedEntityRecord and let the user manually save the whole post at the end when they are done — which is maybe a better UX anyway, no?

Partial save would certainly be a nice feature! I don't we have it today, but you could also pass a custom `__unstableFetch` option and pre-process the http request body before sending it through. It feels hacky, though, so perhaps do that only if it's really important :-)

Reply:
> Also surfacing `useEntityRecord`:

The issue with `useEntityRecord`, at least in my testing, was that it doesn't consider _edited_ entities. That is, toggling `comment_status` on the post entity doesn't change the entity record until the entity is saved.

While that doesn't make it a dealbreaker for the comment-enabling functionality debated in this PR, it would still require some manual synchronisation between the edited entity, the warning UI, and the saving procedure. That's why `useEntityProp` felt better to me in the end. Or am I missing something with `useEntityRecord`, @adamziel?

Reply:
Ah no, you're right @mcsf. The support for edits is coming, but isn't there yet: https://github.com/WordPress/gutenberg/pull/39595

Reply:
Thanks a lot, folks! I'm gonna give that (i.e. `useEntityProp` with post ID and type from block context) a try!

Reply:
Well, that was easy: https://github.com/WordPress/gutenberg/pull/41603/commits/dda7509644ee2b694757b4ad7e72afe38b17644c

Turns out we were already using `useEntityProp` and block context to read the comment status anyway 😬

Reply:
😆 wonderful!